azimuth/dataset_split_manager.py

```diff
@@ -88,22 +89,30 @@ def __init__(
         self._index_path = pjoin(self._base_dataset_path, "index.faiss")
         self._features_path = pjoin(self._base_dataset_path, "features.faiss.npy")
         self._file_lock = pjoin(self._hf_path, f"{name}.lock")
-        self.last_update = -1
         # Load the dataset_split from disk.
-        cached_dataset_split = self._load_dataset_split()
-        if cached_dataset_split is None:
+        self._base_dataset_split_last_update: Time = -1
+        cached_base_dataset_split = self._load_latest_base_dataset_split()
+        if cached_base_dataset_split is None:
             if dataset_split is None:
                 raise ValueError("No dataset_split cached, can't initialize.")
             log.info("Initializing tags", tags=initial_tags)
-            self._base_dataset_split, self._malformed_dataset = self._load_base_dataset_split(
+            self._base_dataset_split, self._malformed_dataset = self._process_base_dataset_split(
                 dataset_split
             )
             self._save_base_dataset_split()
         else:
-            self._base_dataset_split, self._malformed_dataset = cached_dataset_split
+            self._base_dataset_split, self._malformed_dataset = cached_base_dataset_split
         self._prediction_tables: Dict[PredictionTableKey, Dataset] = {}
+        self._prediction_tables_last_update: Dict[PredictionTableKey, Time] = defaultdict(int)
         self._validate_columns()
 
+    @property
+    def last_update(self) -> Time:
+        return max(
+            [self._base_dataset_split_last_update]
+            + list(self._prediction_tables_last_update.values())
+        )
```

Comment:
There seems to be no need for lists here:
```suggestion
        return max(self._base_dataset_split_last_update, *self._prediction_tables_last_update.values())
```

Reply:
Yeah, I was waiting for feedback on that row - I couldn't figure out the best way to do it.
With your suggestion, I get `TypeError: 'float' object is not iterable`, when `self._prediction_tables_last_update` is an empty `dict`.

Reply:
Ah! That makes sense. One list (or tuple) then!
```suggestion
        return max(
            (self._base_dataset_split_last_update, *self._prediction_tables_last_update.values())
        )
```

Reply:
Perfect, that works! I never know when to use the `*`.

Reply:
I like the `*` operator. I'll prepare a short presentation to the team so we all understand it well.